test/parallel/test-net-connect-options-fd.js

```diff
@@ -90,7 +90,7 @@ const forAllClients = (cb) => common.mustCall(cb, CLIENT_VARIANTS);
       assert(handleMap.has(index));
       const oldHandle = handleMap.get(index);
       assert.strictEqual(oldHandle.fd, this._handle.fd);
-      this.write(String(oldHandle.fd));
+      this.end(String(oldHandle.fd));
```

Comment:
@jasnell: I don't understand how this test could succeed before? How does the socket emit 'end' on the server side when it's never ended by client side?

Reply:
Is an `end()` on server side supposed the close the socket? Not sure how that works today? `allowHalfOpen` only seems to apply for `Readable` side and not `Writable`.